### PR TITLE
Feat/fe 84 change channel to private

### PR DESCRIPTION
**Pull request by queen-x-vee**
# Make channel private / (https://linear.app/team-bevel/issue/FRO-84/change-channel-to-private-functionality)

closes issue #2722 ( https://github.com/zurichat/zc_main/pull/2722)
( This could be an existing issue or a linear ticket )

**My PR Closes #2722** or
**My PR Fixes ID-linear_ticket_number_here**

# Changes proposed

## What were you told to do?
 I was asked to make the channel private functionality work

## What did you do?

I created a private channel component where I made my functionality. I also made modifications to the message detail dialog box so the user can choose to make the channel private

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 1]
```diff
@@ -1,10 +1,59 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "../styles/Drop.module.css";
 import ToggleSwitch from "./ToggleSwitch";
 import { BsToggleOff } from "react-icons/bs";
 import GrCircleAlert from "react-icons/gr";
+import axios from "axios";
 
 const DropRoom = props => {
+  //{ updatePrivacy, createChannel } = useChannel(props.data?.room_id);
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:9]: remove comment

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 2]
```diff
@@ -1,10 +1,59 @@
...
+  const handleEvent = ({ target }) => {
+    setState({
+      ...state,
+      [target.name]: target.type === "checkbox" ? target.checked : target.value
+    });
+  };
+
+  const handleSubmission = e => {
+    e.preventDefault();
+    //createChannel(state);
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:21]: comment

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 3]
```diff
@@ -1,10 +1,59 @@
...
+    console.log(state);
+  };
+  const org_id = localStorage.getItem("currentWorkspace");
+  const user = JSON.parse(sessionStorage.getItem("organisations"));
+  const room = window.location.href.split("/").at(6);
+  const BASE_URL = "https://chat.zuri.chat";
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:27]: the base url should already be in an axios utils file

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 4]
```diff
@@ -1,10 +1,59 @@
...
+    axios
+      .get(`${BASE_URL}/api/v1/org/${org_id}/rooms/${room}`)
+      .then(res => {
+        setData(res.data.data);
+      })
+      .catch(e => console.log(e));
+  }, [5]);
+
+  const updatePrivacy = () => {
+    const tina = user?.find(x => x.id === org_id)?.member_id;
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:41]: probably don't use your name 😄

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 5]
```diff
@@ -1,10 +1,59 @@
...
+    if (data !== null) {
+      const newData = { ...data, is_private: JSON.stringify(state.Accept) };
+      console.log(`this is a new ${newData}`);
+      axios
+        .put(
+          `${BASE_URL}/api/v1/org/${org_id}/members/${tina}/rooms/${room}`,
+          newData
+        )
+        .then(res => {
+          console.log(res);
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:51]: don't log

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 6]
```diff
@@ -1,10 +1,59 @@
...
+        })
+        .catch(e => console.log(e));
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:53]: handle the error appropriately use a console error and then display a toast notification to let the user know

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx [hunk 7]
```diff
@@ -57,7 +59,8 @@ function MessageRoomDetailsDialog({
   const toggleDeleteChannel = () => setShowDeleteChannel(!showDeleteChannel);
   const toggleLeaveChannelModal = () =>
     setShowLeaveChannelModal(!showLeaveChannelModal);
-  const toggleArchiveChannel = () => setShowArchiveChannel(!showArchiveChannel);
+  const toggleArchiveChannel = () => setShowArchiveChannel(prev => !prev);
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx:62]: great use of the usestate function

File: packages/ui/src/message-room-details-dialog/auth/useChannel.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+
+const useChannel = data => {
+  let { check } = data;
+  const BASE_API_URL = "https://chat.zuri.chat";
+  const token = sessionStorage.getItem("token");
+  //const member_id = localStorage.getItem("user");
+  const org_id = localStorage.getItem("currentWorkspace");
+  const valentina = localStorage.getItem("room_id");
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/auth/useChannel.js:9]: your name again

File: packages/ui/src/message-room-details-dialog/auth/useChannel.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  const member = JSON.parse(sessionStorage.getItem("user")) || null;
+  const member_id = member.id;
+  localStorage.setItem("isPrivate", check);
+
+  const updatePrivacy = privat =>
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/auth/useChannel.js:14]: spelling error

File: packages/ui/src/message-room-details-dialog/auth/useChannel.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+
+const useChannel = data => {
+  let { check } = data;
+  const BASE_API_URL = "https://chat.zuri.chat";
+  const token = sessionStorage.getItem("token");
+  //const member_id = localStorage.getItem("user");
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/auth/useChannel.js:7]: comment

File: packages/ui/src/message-room-details-dialog/auth/useChannel.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+
+const useChannel = data => {
+  let { check } = data;
+  const BASE_API_URL = "https://chat.zuri.chat";
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/auth/useChannel.js:5]: this url should already be in the app

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  NotificationContainer,
+  NotificationManager
+} from "react-notifications";
+
+const PrivateChannel = ({ closeEdit }) => {
+  const createNotification = type => {
+    return () => {
+      switch (type) {
+        case "info":
+          NotificationManager.info("Info message");
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:15]: did you mean to leave these as template messages?

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+          break;
+        case "success":
+          NotificationManager.success("Success message", "Title here");
```
Comment (ickynavigator) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:18]: also they should be translated

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 14]
```diff
@@ -0,0 +1,209 @@
+import { useState, useCallback, useEffect } from "react";
+import styled from "styled-components";
+import { EditorState, RichUtils, convertToRaw } from "draft-js";
+import Editor from "@draft-js-plugins/editor";
+import createMentionPlugin, {
+  defaultSuggestionsFilter
+} from "@draft-js-plugins/mention";
+import "emoji-mart/css/emoji-mart.css";
+import "!style-loader!css-loader!draft-js-emoji-plugin/lib/plugin.css";
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:9]: you're importing a lot of css files
did you make sure they weren't imported already?

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+import "@draft-js-plugins/mention/lib/plugin.css";
+import "!style-loader!css-loader!@draft-js-plugins/emoji/lib/plugin.css";
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:11]: also check if the packages have minified versions instead

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+            handleKeyCommand={handleKeyCommand}
+            plugins={[emojiPlugin, mentionPlugin]}
+          />
+        </div>
+        <MentionSuggestions
+          open={suggestionsOpen}
+          onOpenChange={onOpenChange}
+          onSearchChange={onSearchChange}
+          suggestions={suggestions}
+          // onAddMention={m => console.log(m)}
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:126]: remove comment

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 17]
```diff
@@ -0,0 +1,209 @@
...
+          editorState={editorState}
+          setEditorState={setEditorState}
+          emojiSelect={<EmojiSelect />}
+          sendMessageHandler={onSendMessage}
+          sendAttachedFileHandler={onAttachFile}
+          sentAttachedFile={sentAttachedFile =>
+            setSentAttachedFile(sentAttachedFile)
+          }
+        />
+        {/* <div>
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:138]: remove comment

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+        </div> */}
+      </InputWrapper>
+    </Wrapper>
+  );
+};
+
+export default MessageInputBox;
+
+const Wrapper = styled.div`
+  // padding: 0 10px;
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:151]: remove commented style

File: packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  background-color: white;
+  width: 100%;
+`;
+
+const InputWrapper = styled.section`
+  border: 1px solid #b0afb0;
+  ${"" /* border: 1px solid hsla(0, 0%, 92%, 1); */}
```
Comment (ickynavigator) [inline packages/zuriUi/src/components/v2/message-editor-input/MessageEditorInput.jsx:161]: remove all commented styles

**Review by ickynavigator — CHANGES_REQUESTED** [review]

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx (hunk 4 above)
Comment (JulianaSau) [reply packages/main/src/components/protected/sidebar/components/DropRoom.jsx:41]: 🤣🤣

**Pull request closed without merging**